cli/.js/src/main/scala/aqua/json/JsonEncoder.scala

```diff
@@ -0,0 +1,118 @@
+package aqua.json
+
+import aqua.types.{ArrayType, BottomType, LiteralType, OptionType, StructType, Type}
+import cats.data.{NonEmptyMap, Validated, ValidatedNec}
+import cats.data.Validated.{invalid, invalidNec, invalidNel, valid, validNec, validNel}
+import cats.syntax.applicative.*
+import cats.syntax.apply.*
+import cats.syntax.flatMap.*
+import cats.syntax.semigroup.*
+import cats.syntax.functor.*
+import cats.syntax.traverse.*
+
+import scala.collection.immutable.SortedMap
+import scala.scalajs.js
+
+object JsonEncoder {
+
+  /* Get widest possible type from JSON arrays. For example:
+  JSON: {
+          field1: [
+                  {
+                    a: "a",
+                    b: [1,2,3],
+                    c: 4
+                  },
+                  {
+                    c: 3
+                  }
+                  ]
+        }
+  There type in array must be { a: ?string, b: []number, c: number
+   */
+  def compareAndGetWidestType(
+    name: String,
+    ltV: ValidatedNec[String, Type],
+    rtV: ValidatedNec[String, Type]
+  ): ValidatedNec[String, Type] = {
+    (ltV, rtV) match {
+      case (Validated.Valid(lt), Validated.Valid(rt)) =>
+        (lt, rt) match {
+          case (lt, rt) if lt == rt => validNec(lt)
+          case (_ @BottomType, ra @ ArrayType(_)) => validNec(ra)
+          case (la @ ArrayType(_), _ @BottomType) => validNec(la)
+          case (lo @ OptionType(lel), rtt) if lel == rtt => validNec(lo)
+          case (ltt, ro @ OptionType(rel)) if ltt == rel => validNec(ro)
+          case (_ @BottomType, rb) => validNec(OptionType(rb))
+          case (lb, _ @BottomType) => validNec(OptionType(lb))
+          case (lst: StructType, rst: StructType) =>
+            val lFieldsSM: SortedMap[String, Type] = lst.fields.toSortedMap
+            val rFieldsSM: SortedMap[String, Type] = rst.fields.toSortedMap
+            (lFieldsSM.toList ++ rFieldsSM.toList)
+              .groupBy(_._1)
+              .view
+              .mapValues(_.map(_._2))
+              .map {
+                case (name, t :: Nil) =>
+                  compareAndGetWidestType(name, validNec(t), validNec(BottomType)).map(t =>
+                    (name, t)
+                  )
+                case (name, lt :: rt :: Nil) =>
+                  compareAndGetWidestType(name, validNec(lt), validNec(rt)).map(t => (name, t))
+                case _ => invalidNec("Unexpected. The list can only have 1 or 2 arguments.")
```

Comment:
I am absolutely sure that this error message will deeply confuse the user, if ever occur. Which list? This list appears only as a part of merging process, it does not exist for the user.